src/stores/useExampleStore.ts

```diff
@@ -1,12 +1,8 @@
 /**
- * This file adheres to our standard for Zustand stores without using `persist`.
+ * This file adheres to our standard for Zustand stores.
  * The state and actions are externalized, promoting modularity and separation of concerns,
  * as outlined in Zustand's best practices: https://zustand.docs.pmnd.rs/guides/practice-with-no-store-actions.
  *
- * We use a namespace in the barrel file (e.g., export * as authStore from "./useAuthStore")
```

Comment:
Now that I started using it, I'm not sure about the barrel. That means everytime we use the store it looks like `authStore.useAuthStore()` which looks a little strange. Typescript will warn you if you try to name two exports the same, so we can rely on that, but I'm open for discussion:

![Screenshot 2024-10-03 at 09 40 33](https://github.com/user-attachments/assets/e0a9f0c3-4701-4d23-b339-2329296a759c)

Reply:
We could name the store hook just `use` under the namespace so usage would be something like:

```
const token = authStore.use(state => state.token)
...
authStore.setToken(newToken)
```

Or even pre-export setters **and getters** like this:

```
export const setToken = (newToken) => useAuthStore.setState({ token: newToken });
export const getToken = () => useAuthStore((state) => state.token)
```

Then usage would be:

```
const token = authStore.getToken()
authStore.setToken(newToken)
```